tools/export/cdt/__init__.py

```diff
@@ -1,15 +1,49 @@
 import re
-
+import os
+import json
+from collections import namedtuple
+from tools.targets import TARGET_MAP
 from os.path import join, exists
 from os import makedirs, remove
 import shutil
 
 from tools.export.makefile import Makefile, GccArm, Armc5, IAR
 
+_eclipse_defs = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), 'cdt_definitions.json')
+
+with open(_eclipse_defs, 'r') as f:
+    _CONFIGS_OPTIONS = json.load(f)
+
+supported_launches = ['debug', 'program']
+
 class Eclipse(Makefile):
     """Generic Eclipse project. Intended to be subclassed by classes that
     specify a type of Makefile.
     """
+    def get_target_config(self, ctx, configuration):
```

Comment:
This method does not use self. Please: 
```suggestion
    @staticmethod
    def get_target_config(ctx, configuration):
```

Reply:
@theotherjimmy,
It is used right here:
tgt = TARGET_MAP[self.target]

Reply:
I must have missed that. Thanks for pointing it out.